File: src/numeric/max/max.php
<?php
/*
 *  This program takes a list of values and outputs the maximum value of those
 */

/**
 * Finds the maximum value in a list of numbers.
 * 
 * @param list The list of numbers to find the maximum value from.
 * @return int The maximum value in the list.

 */
function max_list($list) {
    $current = 0;
    foreach ($list as $value) {
        if ($value > $current) {
            $current = $value;
        }
    }
    return $current;
}

$list = [15, 10];
if ($argc > 1) {
    array_shift($argv);
    $list = $argv;
}

$maxValue = max_list($list);

echo "values: ";
print_r($list);
echo "max: $maxValue\n";
?>

File: src/numeric/max/max.rb
=begin
    This program finds the max value of a sequence of values
=end

# Find the max value inside a list
def max_list(list)
    current = 0
    list.each do |value|
        if value > current
            current = value
        end
    end
    return current
end

list = [15, 10]
if ARGV.length > 0
    list = ARGV.map { |arg| arg.to_i }
end

maxValue = max_list(list)

puts "values: #{list}"
puts "max: #{maxValue}"
